src/Raven.Server/Documents/Handlers/Processors/Streaming/StreamingHandlerProcessorForGetStreamQuery.cs

```diff
@@ -0,0 +1,146 @@
+﻿using System;
+using System.IO;
+using System.Net;
+using System.Net.Http;
+using System.Threading.Tasks;
+using JetBrains.Annotations;
+using Microsoft.AspNetCore.Http;
+using Raven.Client.Exceptions.Documents.Indexes;
+using Raven.Server.Documents.Indexes;
+using Raven.Server.Documents.Queries;
+using Raven.Server.NotificationCenter;
+using Raven.Server.ServerWide;
+using Raven.Server.ServerWide.Context;
+using Sparrow.Json;
+
+namespace Raven.Server.Documents.Handlers.Processors.Streaming
+{
+    internal class StreamingHandlerProcessorForGetStreamQuery : AbstractStreamingHandlerProcessorForGetStreamQuery<DatabaseRequestHandler, DocumentsOperationContext>
+    {
+        protected QueryOperationContext _queryContext;
+
+        public StreamingHandlerProcessorForGetStreamQuery([NotNull] DatabaseRequestHandler requestHandler, HttpMethod method) : base(requestHandler, method)
+        {
+        }
+
+        protected override RequestTimeTracker GetTimeTracker()
+        {
+            return new RequestTimeTracker(HttpContext, Logger, RequestHandler.Database.NotificationCenter, RequestHandler.Database.Configuration, "StreamQuery",
+                doPerformanceHintIfTooLong: false);
+        }
+
+        protected override ValueTask<BlittableJsonReaderObject> GetDocumentData(DocumentsOperationContext context, string fromDocument)
+        {
+            using (context.OpenReadTransaction())
+            {
+                var document = RequestHandler.Database.DocumentsStorage.Get(context, fromDocument);
+                return ValueTask.FromResult(document?.Data.Clone(context));
+            }
+        }
+
+        protected override IDisposable AllocateContext(out DocumentsOperationContext context)
```

Comment:
better to out QueryOperationContext and in the sharding one just set needsServerContext to false?

Reply:
The `QueryOperationContext` ctor always requires a `DocumentDatabase` though?

Reply:
true, nvm